riscv-pages/src/page.rs

```diff
@@ -38,6 +38,22 @@ impl PageSize {
     pub const fn num_4k_pages(val: u64) -> u64 {
         (val + PageSize::Size4k as u64 - 1) / (PageSize::Size4k as u64)
     }
+    /// Return the number of bits to shift to get one page.
+    /// i.e. 4096 == 2^12 == 1 << 12
+    pub fn page_shift(&self) -> u64 {
+        match self {
+            PageSize::Size4k => 12,
+            PageSize::Size2M => 21,
+            PageSize::Size1G => 30,
+            PageSize::Size512G => 39,
+        }
+    }
+
+    /// Return the number of pages of this `PageSize` that will fit in `val`.
+    /// This rounds down the number of pages, unlike num_4k_pages() which rounds up.
+    pub fn num_pages_contained_in(&self, val: u64) -> u64 {
```

Comment:
nit: would val be better names size_bytes? It took me a minute to figure out what it meant.

Reply:
fixed. It's not the right function name, but I'm not sure what to call it.